src/main/java/com/example/auctionapp/model/User.java

```diff
@@ -50,6 +71,31 @@ public User(String firstName, String lastName, String email, String password, Ro
         this.image = image;
     }
 
+    public User(
+            String firstName,
+            String lastName,
+            String email,
+            String password,
+            Role role,
+            Image image,
+            String phoneNumber,
+            LocalDate birthDate,
+            GenderEnum gender,
+            Address address,
+            CardInformation cardInformation) {
```

Comment:
A large constructor often means that the class can be split into parts, user class is slowly become more and more complex, so it would be good to extract user personal information to one class, and user account information to an another class.
This would require changes in the db of course.

Reply:
something like UserAccount, UserInformation, UserLoginInformation tables. Where UserAccount would be the main user table